src/lib/form-field/form-field.ts

```diff
@@ -148,6 +148,11 @@ export class MatFormField extends _MatFormFieldMixinBase
     return this._appearance || this._defaultOptions && this._defaultOptions.appearance || 'legacy';
   }
   set appearance(value: MatFormFieldAppearance) {
+    // If we're switching to `outline` from another appearance, we have to recalculate the gap.
+    if (value !== this._appearance && value === 'outline') {
```

Comment:
should we add a test for this?

Reply:
did you try just doing a real async test?

Reply:
I got some time to take a better look and it turns out that my test wasn't working because I didn't have a label. I've pushed a test that fails when we expect it to.